### PR TITLE
Enable hybrid build-runtime archive configuration

### LINKED ISSUE #30: Support 'SPA Mode' with archives.json-based config.
Have an option where the the archives.js is instead an archives.json, and the entire site can be loaded as an SPA?

This should then have all of the settings, such as title, also be loaded form archives.json

Perhaps this would also disable sitemap.xml generation and anything else that requires a separate page.

Can we support both current mode and SPA-mode with simple toggle?

This would allow taking an existing published site as a template, and simply replacing `archives.json` to get a new site, and publishing to a new static site (eg. on IPFS).

### PR DESCRIPTION
**Pull request by SuaYoo**
- Add option to configure archive data at website runtime with `runtimeOnlyArchives`.
- Refactor UI based on `archivePages` into lit components

Example of runtime configuration: https://github.com/webrecorder/web-replay-gen/tree/config-client-runtime/examples/runtime-only-archives#readme
Documentation of configuration updates: https://github.com/webrecorder/web-replay-gen/tree/config-client-runtime#configuration

Resolves #30.

### Manual testing
1. Run `npm i` to get new dependencies
2. Remove `archives` from `wrg-config.json`
3. Specify a path as `runtimeOnlyArchives` in `wrg-config.json`
4. Run `npm run serve`. Verify archives load as expected
5. Update archive JSON in `_site` and reload page without rebuilding. Verify archives update as expected

### Follow-ups
- [ ] https://github.com/webrecorder/web-replay-gen/issues/32
- [x] Decrease size of build (unnecessary node_modules?)

### CODE REVIEW

File: src/_includes/layout.njk [hunk 1]
```diff
@@ -7,11 +7,118 @@
...
+        // pre-rendered components will come to life and become interactive.
+        //
+        // You may also prefer to bundle your components into fewer JS modules.
+        // See https://lit.dev/docs/tools/production/#building-with-rollup for
+        // more details.
+        {% for path in wc.filePaths %}
+        import('{{ path }}');
+        {% endfor %}
+
+        {% if archiveConfig.runtimeOnlyArchives %}
```
Comment (SuaYoo) [inline src/_includes/layout.njk:109]: This is where `runtimeOnlyArchives` gets loaded; the rest of the changes in this file is mostly pulled from https://github.com/lit/lit/tree/main/packages/labs/eleventy-plugin-lit

File: README.md [hunk 2]
```diff
@@ -175,6 +175,31 @@ The default behavior is to list Web Archive files in the `archives` directory. W
...
+| Key                   | Default Value | Value Type |                                                        |
+| --------------------- | ------------- | ---------- | ------------------------------------------------------ |
+| `runtimeOnlyArchives` | `undefined`   | `string`   | Path to JSON file with `archives`, relative to `_site` |
+
+By default, the generator configures the location of your archives at build-time (i.e. when you run `npm run build`) in order to pre-process data and render static HTML based on that data. However, you may have a use case where you need to check archive locations every time the website loads in the browser. Setting `runtimeOnlyArchives` enables you to do things like configure and update archive data without redeploying your entire website.
+
+Caveats:
+
+- The generated sitemap will no longer list a page per archive.
+- Top-level `archives` config must be a JSON file relative to the generated `_site` directory. For example, if you have a JSON file in `_site/data/archive-list.json`, `archives` should be `./data/archive-list.json`.
```
Comment (ikreymer) [inline README.md:196]: This means that any local files listed in `archives` config in `./data-archive-list.json` should be relative to `_site`, right?

**Comment by ikreymer** [comment]
Nice! Currently getting some errors when running, probably all minor. 

With: 
```
{
  "runtimeOnlyArchives": "./__generated__/archives.json"
}
```
got a build time error (TypeError: Cannot read properties of undefined (reading 'runtimeOnlyArchives')

Was building with this, and looks like the .json is loading, but then disappears?
```
{
  "site": {},
  "runtimeOnlyArchives": "./__generated__/archives.json"
}
```

**Comment by ikreymer** [comment]
Perhaps instead of having to edit in the config and to switch between either `archives` or `runtimeOnlyArchives` in the config, we instead have two different builds the 'static' mode and the 'dynamic' mode with client-side loading, eg having `yarn build` and `yarn build-dynamic` and `yarn serve` `yarn serve-dynamic`? That way, could generate both from the same config and always have the generated json there?

File: README.md (hunk 2 above)
Comment (SuaYoo) [reply README.md:196]: Yep, I'll clarify

**Comment by SuaYoo** [comment]
> Was building with this, and looks like the .json is loading, but then disappears?

This is fixed now, it was getting overwritten on config and src changes.

**Review by ikreymer — APPROVED** [review]

**Pull request merged**